cmd/horcrux/cmd/config.go

```diff
@@ -548,28 +551,57 @@ func (c *DiskConfig) CosignerPeers() (out []signer.CosignerConfig) {
 	return
 }
 
+func (c *DiskConfig) KeyAndThresholdSigner(logger tmlog.Logger) (signer.CosignerKey, signer.ThresholdSigner, error) {
+	switch c.CosignerConfig.SignerType {
+	case "hsm", "HSM":
+		logger.Info("Cosigning with HSM")
+		return signer.CosignerKey{}, signer.NewThresholdSignerHSM(), nil
+	default:
+
+		keyFilePath := config.keyFilePath(true)
+		if _, err := os.Stat(keyFilePath); os.IsNotExist(err) {
+			return signer.CosignerKey{}, nil, fmt.Errorf("private key share doesn't exist at path(%s)", keyFilePath)
+		}
+		key, err := signer.LoadCosignerKey(keyFilePath)
+		if err != nil {
+			return signer.CosignerKey{}, nil, fmt.Errorf("error reading cosigner key: %s", err)
+		}
+		logger.Info("Cosigning with soft key",
+			"file", keyFilePath,
+			"id", key.ID,
+			"threshold", c.CosignerConfig.Threshold,
+			"total", c.CosignerConfig.Shares,
+		)
+		return key, signer.NewThresholdSignerSoft(
+			key,
+			c.CosignerConfig.Threshold,
+			c.CosignerConfig.Shares,
+		), nil
+	}
+}
+
 type CosignerPeer struct {
 	ShareID int    `json:"share-id" yaml:"share-id"`
 	P2PAddr string `json:"p2p-addr" yaml:"p2p-addr"`
 }
 
-func validateCosignerPeers(peers []CosignerPeer, shares int) error {
+func validateCosignerPeers(peers []CosignerPeer, shares uint8) error {
 	// Check IDs to make sure none are duplicated
 	if dupl := duplicatePeers(peers); len(dupl) != 0 {
 		return fmt.Errorf("found duplicate share IDs in args: %v", dupl)
 	}
 
 	// Make sure that the peers' IDs match the number of shares.
 	for _, peer := range peers {
-		if peer.ShareID < 1 || peer.ShareID > shares {
+		if peer.ShareID < 1 || peer.ShareID > int(shares) {
```

Comment:
Because ShareID, peers etc is int. 

Would be good in the future to decide which int type to use. And I assume that 
The current way is not a great design choice. Maybe should be added to issues or some kind of roadmap?